test/integration_test.js

```diff
@@ -178,51 +194,223 @@ describe("Integration tests", () => {
+            });
+          });
+
+          it("should put local database in the expected state", () => {
+            return tasks.list({order: "title"})
+              .then(res => res.data.map(record => ({
+                title: record.title,
+                done: record.done,
+                _status: record._status,
+              })))
+              .should.become([
+                {title: "task1", _status: "synced", done: true},
+                {title: "task2", _status: "synced", done: false},
+                {title: "task3", _status: "synced", done: true},
+                // For CLIENT_WINS strategy, local record is marked as synced
+                {title: "task4-local", _status: "synced", done: false},
+              ]);
+          });
         });
 
-        it("should not have updated anything", () => {
-          expect(syncResult.updated).to.have.length.of(0);
+        describe("SERVER_WINS strategy", () => {
+          beforeEach(() => {
+            return testSync(testData, {
+              strategy: Kinto.syncStrategy.SERVER_WINS
+            }).then(res => syncResult = res);
+          });
+
+          it("should not have an ok status", () => {
+            expect(syncResult.ok).eql(true);
+          });
+
+          it("should contain no errors", () => {
+            expect(syncResult.errors).to.have.length.of(0);
+          });
+
+          it("should have a valid lastModified value", () => {
+            expect(syncResult.lastModified).to.be.a("number");
+          });
+
+          it("should have no incoming conflict listed", () => {
+            expect(syncResult.conflicts).to.have.length.of(0);
+          });
+
+          it("should not skip records", () => {
+            expect(syncResult.skipped).to.have.length.of(0);
+          });
+
+          it("should not import anything", () => {
+            expect(syncResult.created).to.have.length.of(0);
+          });
+
+          it("should publish resolved conflict using remote version", () => {
+            expect(syncResult.published).to.have.length.of(1);
+            expect(cleanRecord(syncResult.published[0])).eql({
+              id: conflictingId,
+              title: "task4-remote",
+              done: true,
+            });
+          });
+
+          it("should update conflict with remote version", () => {
+            expect(syncResult.updated).to.have.length.of(1);
+            expect(cleanRecord(syncResult.updated[0])).eql({
+              id: conflictingId,
+              title: "task4-remote",
+              done: true,
+            });
+          });
+
+          it("should list resolved records", () => {
+            expect(syncResult.resolved).to.have.length.of(1);
+            expect(cleanRecord(syncResult.updated[0])).eql({
+              id: conflictingId,
+              title: "task4-remote",
+              done: true,
+            });
+          });
+
+          it("should put local database in the expected state", () => {
```

Comment:
I would add a test to make sure conflict resolution is not send to the remote server (which is already up-to-date since we've selected the remote version)